libs/domain/picking/src/components/user-profile/user-profile.component.ts

```diff
@@ -1,39 +1,46 @@
 import { AuthService } from '@spryker-oryx/auth';
-import { AppRef, InjectionPlugin } from '@spryker-oryx/core';
-import { resolve } from '@spryker-oryx/di';
+import { AppRef } from '@spryker-oryx/core';
+import { INJECTOR, resolve } from '@spryker-oryx/di';
 import { SyncSchedulerService } from '@spryker-oryx/offline';
+import { OfflineDataPlugin } from '@spryker-oryx/picking/offline';
 import { RouterService } from '@spryker-oryx/router';
 import { CLOSE_EVENT } from '@spryker-oryx/ui/modal';
-import { asyncState, i18n, valueType } from '@spryker-oryx/utilities';
+import { i18n, signal, signalAware } from '@spryker-oryx/utilities';
 import { html, LitElement, TemplateResult } from 'lit';
 import { state } from 'lit/decorators.js';
 import { when } from 'lit/directives/when.js';
-import { delay, switchMap, tap } from 'rxjs';
-import { OfflineDataPlugin } from '../../../offline/data-plugin';
+import { delay, tap } from 'rxjs';
 import { userProfileComponentStyles } from './user-profile.styles';
 
+signalAware();
 export class UserProfileComponent extends LitElement {
   static styles = userProfileComponentStyles;
 
   protected routerService = resolve(RouterService);
   protected authService = resolve(AuthService);
   protected syncSchedulerService = resolve(SyncSchedulerService);
 
-  @state()
-  protected loading: boolean | null = null;
+  protected injector = resolve(INJECTOR);
+  protected injectorDataPlugin =
+    resolve(AppRef).requirePlugin(OfflineDataPlugin);
 
-  @asyncState()
-  protected currentRoute = valueType(this.routerService.currentRoute());
+  @state()
+  protected hasPendingSyncs: boolean | null = null;
 
-  @asyncState()
-  protected route = valueType(this.routerService.route());
+  @state()
+  protected loading: boolean | null = null;
 
-  @asyncState()
-  protected pendingSyncs = valueType(this.syncSchedulerService.hasPending());
+  protected route = signal(this.routerService.route());
+  protected pendingSyncs = signal(
+    this.syncSchedulerService
+      .hasPending()
+      .pipe(tap((pending) => (this.hasPendingSyncs = pending)))
+      .subscribe()
```

Comment:
@dunqan can you help on this one? it's the case we talk last week, that it's forcing to subscribe on the component.
This is something to avoid completely, or it could be necessary in some cases?